File: homework1/cpp/friend.cpp
#include <iostream>
#include <string>
#include "person.h"		/* Person 클래스의 정의문을 가져옴 */
#include "friend.h"		/* Friend 클래스의 정의문을 가져옴 */
#include <cstdio>

Friend :: Friend(string& fn, string& ln, int& pn, int& a) : Person(fn, ln, pn)	//  Friend 생성자
{
    setAge(a);
}

void Friend :: setAge(int a)
{
    age = a;
}

int Friend :: getAge(void)
{
    return age;
}

void Friend :: print(void)
{
   	int first = getPhoneNumber()/100000000;
   	int second = (getPhoneNumber() - (first)*100000000)/10000;
   	int third = (getPhoneNumber() - (first)*100000000 - (second)*10000);

	cout << getFirstName() << " " << getLastName() << "_";
	printf("0%d-%04d-%04d_", first, second, third);
	cout << getAge() << endl;
}

File: homework1/java/Friend.java
class Friend extends Person {
    public Friend(String fn, String ln, int pn, int a)
    {
        super(fn, ln, pn);
        setAge(a);
    }


    /* set & get */
    public void setAge(int a)
    {
        age = a;
    }
    public int getAge()
    {
        return age;
    }


    /* print */
    public void print()
    {
    	int first = getPhoneNumber()/100000000;
    	int second = (getPhoneNumber() - (first)*100000000)/10000;
    	int third = (getPhoneNumber() - (first)*100000000 - (second)*10000);
    	
        System.out.print(getFirstName() + " " + getLastName() + "_");
        System.out.printf("0%d-%04d-%04d_", first, second, third);
        System.out.println(getAge());
    }


    /* data */
    private int age;
}
